client/components/Profile.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Loader from "../components/Loader.jsx"
+
+const profileSubs = new SubsManager();
+
+const Profile = React.createClass({
+  mixins: [ReactMeteorData],
+
+  getInitialState: function() {
+    return {
+      nicknameChanged: false
+    };
+   },
+
+  getMeteorData () {
+    const usersHandle = profileSubs.subscribe("users");
+    let user = {};
+
+    if (usersHandle.ready()) {
+      user = Meteor.users.findOne(this.props.userProfile);
+    }
+    return {
+      usersHandle,
+      user
+    }
+  },
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const nickname = ReactDOM.findDOMNode(this.refs.nickname).value.trim();
+    if (nickname.length === 0) return;
+    Meteor.call("user.changeNickname", nickname, (error, result) => {
+      if (!error) {
+        this.setState({nicknameChanged: true})
+      }
+    });
+  },
+
+  componentDidMount() {
+    if (this.data.usersHandle.ready() && this.props.userProfile == Meteor.userId()) {
+      ReactDOM.findDOMNode(this.refs.nickname).focus();
+      $('.ui.form')
+        .form({
+          fields: {
+            nickname: {
+              identifier: 'nickname',
+              rules: [
+                {
+                  type   : 'empty',
+                  prompt : 'Please enter a valid nickname'
+                }
+              ]
+            }
+          }
+        });
+    }
+  },
+
+  render() {
+    if (!this.data.usersHandle.ready()) {
```

Comment:
Do not put reactive data sources in render, it wont work. Do all the reactive stuff, such as ready handles, in getMeteorData, and return a userDataReady variable from it instead.